packages/core/src/utils.ts

```diff
@@ -446,6 +446,43 @@ function deepEqualsObject(
   return true;
 }
 
+/**
+ * Checks if object2 includes all fields and values of object1.
+ * It doesn't matter if object2 has extra fields.
+ * @param object1 The first object.
+ * @param object2 The second object.
+ * @returns True if object2 includes all fields and values of object1.
+ */
+export function deepIncludes(object1: unknown, object2: unknown): boolean {
+  if (isEmpty(object1)) {
+    return true;
+  }
+  if (isEmpty(object2)) {
+    return false;
+  }
+  if (Array.isArray(object1) && Array.isArray(object2)) {
```

Comment:
Minor style nit, but you could compact this whole block of code and reduce visual noise by making the set of conditions one big `if / else if`